docker/compose.yaml

```diff
@@ -0,0 +1,161 @@
+version: '3'
+
+volumes:
+  node_modules:
+  graph_modules:
+  postgres_storage:
+  ipfs_staging:
+  ipfs_data:
+  subgraph.yaml:
+
+
+services:
+  install:
+    image: ghcr.io/hypercerts-org/node-dev-18:1.0
+    working_dir: /usr/src/app
+    command: yarn install --non-interactive --frozen-lockfile
+    volumes:
+      - ../:/usr/src/app
+      - node_modules:/usr/src/app/node_modules
+  localchain:
+    image: ghcr.io/hypercerts-org/node-dev-18:1.0
+    ports:
+      - "${LOCALCHAIN_PORT}:8545"
+    working_dir: /usr/src/app/contracts
+    command: yarn hardhat node
```

Comment:
As a sanity check: In the past I got timestamp issues running hardhat, graph and IPFS in the same Docker env. The solution was purging the volumes after shutdown, iirc because the Graph kept blocknumbers in the volume, hardhat reboots to 0.

Have you tried multiple stop-starts? I.e. is this issue still relevant?

Reply:
As long as all services are brought down then you can cleanly restart